analyzers/tests/SonarAnalyzer.UnitTest/Common/RuleTest.cs

```diff
@@ -159,9 +168,13 @@ public void AllRules_SonarWayTagPresenceMatchesIsEnabledByDefault()
             {
                 if (IsSecurityHotspot(diagnostic))
                 {
-                    // Security hotspots are enabled by default, but they will report issues only
-                    // when their ID is contained in SonarLint.xml
+                    // Security hotspots are enabled by default, but they will report issues only when their ID is contained in SonarLint.xml
+                    // Rule activation is done in DiagnosticDescriptorBuilder.WithNotConfigurable() to prevent rule supression and deactivation.
                     diagnostic.IsEnabledByDefault.Should().BeTrue($"{diagnostic.Id} should be enabled by default");
+                    if (IsDeprecated(diagnostic))
+                    {
+                        System.Diagnostics.Debugger.Break();
```

Comment:
Is this still needed? it looks to me useful only for debugging

Reply:
Not at all 🤦‍♂️

Reply:
"Don't use debugger features in production" could be a new rule :)